File: 824.山羊拉丁文.py
#
# @lc app=leetcode.cn id=824 lang=python
#
# [824] 山羊拉丁文
#

# @lc code=start
class Solution(object):

    def toGoatLatin(self, S):
        """
        :type S: str
        :rtype: str
        """
        VOWEL = 'aeiouAEIOU'
        ret = ''
        not_vowel = ''
        space = 0
        first_char = True
        for c in S:
            if first_char and c not in VOWEL:
                not_vowel = c
                first_char = False
            elif c == ' ':
                space += 1
                if not_vowel:
                    ret += not_vowel
                    not_vowel = ''
                ret +=  'ma' + 'a' * space + ' '
                first_char = True
            else:
                ret += c
                first_char = False
        else:
            space += 1
            if not_vowel:
                ret += not_vowel
                not_vowel = ''
            ret +=  'ma' + 'a' * space
        return ret
        
# @lc code=end

File: 824.山羊拉丁文.go
/*
 * @lc app=leetcode.cn id=824 lang=golang
 *
 * [824] 山羊拉丁文
 */

// @lc code=start
// package leetcode
import "strings"
var charMap = map[byte]bool{
	'a': true,
	'e': true,
	'i': true,
	'o': true,
	'u': true,
	'A': true,
	'E': true,
	'I': true,
	'O': true,
	'U': true,
}

func toGoatLatin(sentence string) string {
	words := strings.Split(sentence, " ")
	result := make([]string, len(words))
	for i, word := range words {
		tmp := []byte(word)
		if _, ok:= charMap[tmp[0]]; ok{
			result[i] = word + "ma"
		}else {
			result[i] = word[1:] + word[0:1] + "ma"
		}
		result[i] = result[i] + strings.Repeat("a", i+1)
	}
	return strings.Join(result, " ")
}
// @lc code=end

